plugin-dev/Source/Sentry/Private/Android/Java/SentryBridgeJava.java

```diff
@@ -89,6 +94,24 @@ public Double sample(SamplingContext samplingContext) {
 		});
 	}
 
+	private static void checkForUnrealException(SentryEvent event) {
+		if (event.getTags().containsKey("sentry_unreal_exception")) {
+			SentryException exception = event.getUnhandledException();
+			if (exception != null) {
+				exception.setType(event.getTag("sentry_unreal_exception_type"));
+				exception.setValue(event.getTag("sentry_unreal_exception_message"));
+				SentryStackTrace trace = exception.getStacktrace();
+				int numFramesToSkip = Integer.parseInt(event.getTag("sentry_unreal_exception_skip_frames"));
+				List<SentryStackFrame> frames = trace.getFrames();
+				trace.setFrames(frames.subList(0, frames.size() - numFramesToSkip));
+			}
+			event.removeTag("sentry_unreal_exception_type");
+			event.removeTag("sentry_unreal_exception_message");
+			event.removeTag("sentry_unreal_exception_skip_frames");
+			event.removeTag("sentry_unreal_exception");
```

Comment:
We really have to reevaluate the use of tags in the SDK. As a general rule, the SDK should not set any tags and add that information in contexts instead. Relay then handles those and elevates specific ones to tags.